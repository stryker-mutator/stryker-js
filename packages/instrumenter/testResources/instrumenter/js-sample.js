function add(a, b) {
  return a + b;
}

add(40, 2);
