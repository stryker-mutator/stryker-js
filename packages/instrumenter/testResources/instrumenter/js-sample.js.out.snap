// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`instrumenter integration should be able to instrument a simple js file 1`] = `
"var __global_69fa48 = function (g) {
  g.__mutantCoverage__ = g.__mutantCoverage__ || {
    static: {},
    perTest: {}
  };

  g.__coverMutant__ = g.__coverMutant__ || function () {
    var c = g.__mutantCoverage__.static;

    if (g.__currentTestId__) {
      c = g.__mutantCoverage__.perTest[g.__currentTestId__] = g.__mutantCoverage__.perTest[g.__currentTestId__] || {};
    }

    var a = arguments;

    for (var i = 0; i < a.length; i++) {
      c[a[i]] = (c[a[i]] || 0) + 1;
    }
  };

  return g;
}(new Function(\\"return this\\")());

function add(a, b) {
  switch (__global_69fa48.__activeMutant__) {
    case 0:
      {}
      break;

    default:
      __global_69fa48.__coverMutant__(0);
      {
        return __global_69fa48.__activeMutant__ === 1 ? a - b : (__global_69fa48.__coverMutant__(1), a + b);
      }
      break;
  }
}

add(40, 2);"
`;
